### PR TITLE
[openshift-saas-deploy] add support for automated promotions

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-2244

depends on https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/13902

this PR adds support for automated promotions for saas file targets defining `auto: true` in the `promotion` section.
this works by sending messages to the SQS gateway to promote a change and consumed by the MR consumer integration.

### CODE REVIEW

File: reconcile/utils/mr/auto_promoter.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                new_content = '---\n'
+                new_content += yaml.dump(content, Dumper=yaml.RoundTripDumper)
+                msg = f'auto promote {commit_sha} in {saas_file_path}'
+                gitlab_cli.update_file(branch_name=self.branch,
+                                       file_path=saas_file_path,
+                                       commit_message=msg,
+                                       content=new_content)
+
+        if not changes:
+            self.cancel('Saas files are up to date. Nothing to do.')
```
Comment (apahim) [inline reconcile/utils/mr/auto_promoter.py:67]: if, after `self.process()`, there's no diff, no MR will be created anyway

Comment (maorfr) [reply reconcile/utils/mr/auto_promoter.py:67]: where does that happen? i've seen cluster updates MRs created empty.

Comment (apahim) [reply reconcile/utils/mr/auto_promoter.py:67]: https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/mr/base.py#L129-L135

Comment (apahim) [reply reconcile/utils/mr/auto_promoter.py:67]: yep... lgtm!

**Review by apahim — APPROVED** [review]
lgtm

File: reconcile/utils/mr/auto_promoter.py (hunk 1 above)
Comment (maorfr) [reply reconcile/utils/mr/auto_promoter.py]: removed this section

**Pull request merged**